core/src/main/resources/lib/form/class-entry.jelly

```diff
@@ -0,0 +1,43 @@
+<!--
+The MIT License
+
+Copyright (c) 2014, Google, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define"
+         xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+  <st:documentation>
+    Invisible &lt;f:entry> type for embedding a descriptor's $class field.
+    <st:attribute name="clazz">
+      The describable class that we are instantiating via structured form submission.
+    </st:attribute>
+    <st:attribute name="descriptor">
+      The descriptor of the describable that we are instantiating via
+      structured form submission.  Mutually exclusive with clazz.
+    </st:attribute>
+  </st:documentation>
+  <j:set var="clazz" value="${attrs.clazz ?: attrs.descriptor.clazz.name}" />
+  <f:invisibleEntry>
+    <!-- Legacy: Remove once plugins have been staged onto $class -->
+    <input type="hidden" name="stapler-class" value="${clazz}" />
```

Comment:
FWIW I still have a mild preference for handling this in `RequestImpl.getSubmittedForm` so that the deprecated attributes do not need to appear in Jelly, rendered forms, or form submission requests. Just moves the compatibility code out of the way of regular development.